crypto/src/openssl.rs

```diff
@@ -0,0 +1,158 @@
+        let md = Self::get_digest(algs);
+        let args = [
+            &KdfArgument::KbMode(KdfKbMode::Counter),
+            &KdfArgument::Mac(KdfMacType::Hmac(md)),
+            &KdfArgument::KbInfo(measurement_digest),
+            &KdfArgument::Salt(info),
+            &KdfArgument::Key(&measurement_digest),
+        ];
+
+        perform_kdf(KdfType::KeyBased, &args, md.size()).ok()
+    }
+
+    fn derive_ecdsa_pub(
+        algs: AlgLen,
+        cdi: &Self::Cdi,
+        label: &[u8],
+        info: &[u8],
+        pub_x: &mut [u8],
+        pub_y: &mut [u8],
+    ) -> Option<()> {
+        let md = Self::get_digest(algs);
+        let nid = Self::get_curve(algs);
+
+        let args = [
+            &KdfArgument::KbMode(KdfKbMode::Counter),
+            &KdfArgument::Mac(KdfMacType::Hmac(md)),
+            &KdfArgument::KbInfo(label),
+            &KdfArgument::Salt(info),
+            &KdfArgument::Key(cdi),
+        ];
+
+        // Generate public key
+        let priv_bn = BigNum::from_slice(
+            perform_kdf(KdfType::KeyBased, &args, md.size())
+                .unwrap()
+                .as_slice(),
+        )
+        .unwrap();
+
+        let group = EcGroup::from_curve_name(nid).unwrap();
+
+        let mut pub_point = EcPoint::new(&group).unwrap();
+        let mut bn_ctx = BigNumContext::new().unwrap();
+        pub_point
+            .mul_generator(&group, &priv_bn, &mut bn_ctx)
+            .unwrap();
+
+        let mut x = BigNum::new().unwrap();
+        let mut y = BigNum::new().unwrap();
+        pub_point
+            .affine_coordinates(&group, &mut x, &mut y, &mut bn_ctx)
+            .unwrap();
+
+        let point = EccPoint {
+            x: x.to_vec_padded((group.order_bits() / 8).try_into().unwrap())
+                .unwrap(),
+            y: y.to_vec_padded((group.order_bits() / 8).try_into().unwrap())
+                .unwrap(),
+        };
+
+        pub_x.copy_from_slice(point.x.as_slice());
+        pub_y.copy_from_slice(point.y.as_slice());
+        Some(())
+    }
+
+    fn ecdsa_sign_with_alias(
+        algs: AlgLen,
+        digest: &[u8],
+        sig_r: &mut [u8],
+        sig_s: &mut [u8],
+    ) -> Option<()> {
+        let nid = Self::get_curve(algs);
+        let priv_bytes = vec![0u8; algs.size()];
+        let group = EcGroup::from_curve_name(nid).ok()?;
+        let priv_bn = BigNum::from_slice(priv_bytes.as_slice()).ok()?;
+
+        let mut pub_point = EcPoint::new(&group).ok()?;
+        let bn_ctx = BigNumContext::new().ok()?;
+        pub_point.mul_generator(&group, &priv_bn, &bn_ctx).unwrap();
+
+        let ec_priv = EcKey::from_private_components(&group, &priv_bn, &pub_point).ok()?;
+
+        let sig = EcdsaSig::sign::<Private>(digest, &ec_priv).ok()?;
+        sig_r.copy_from_slice(sig.r().to_vec().as_slice());
+        sig_s.copy_from_slice(sig.s().to_vec().as_slice());
+        Some(())
+    }
+}
+
+struct EccPoint {
```

Comment:
Can probably ditch this struct since we aren't returning it anymore

Reply:
Done